### PR TITLE
added Youtube extension

### PR DESCRIPTION
**Pull request by kevinpapst**
See #216

- Moves the YouTube extension to the official repository, to prevent multiple forks
- Removed my other two extensions, as they do not work any longer
- Commits from the currently linked fork should be added in a second PR

**Dear Reviewer** 
I do not have a running FreshRSS installation for testing. The 8+ year old codebase was upgraded blindly to latest FreshRSS and "modern" PHP style, especially the code regarding user configuration handling needs testing. In other words: install before merge 😁

### CODE REVIEW

File: xExtension-YouTube/extension.php [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+     * This function is called by FreshRSS when the configuration page is loaded, and when configuration is saved.
+     *  - We save configuration in case of a post.
+     *  - We (re)load configuration in all case, so they are in-sync after a save and before a page load.
+     */
+    public function handleConfigureAction(): void
+    {
+        $this->registerTranslates();
+
+        if (Minz_Request::isPost()) {
+            FreshRSS_Context::userConf()->yt_player_height = Minz_Request::paramInt('yt_height');
```
Comment (kevinpapst) [inline xExtension-YouTube/extension.php:251]: @Alkarex as I am upgrading code blindly here, is `_attribute()` there correct way to set configuration?

https://github.com/FreshRSS/FreshRSS/blob/354a5e83888cebe89843b754d0936f3633323cef/app/Models/UserConfiguration.php#L129-L131

File: xExtension-YouTube/extension.php [hunk 2]
```diff
@@ -180,7 +180,7 @@ public function getHtmlPeerTubeContentForLink(FreshRSS_Entry $entry, string $lin
     }
 
     /**
-     * Returns an HTML <iframe> for a given URL for the configured width and height, with content ignored, appended or formated.
```
[the remark this answers is not in the document]
Comment (kevinpapst) [reply xExtension-YouTube/extension.php]: Never seen that in any repository, weird and funny at the same time

**Comment by Alkarex** [comment]
Help to test welcome (I am not a user myself), and optionally to finalise the light code upgrade  (e.g. `[ ]` instead of `array()`, `<?= $x ?>` instead of `<?php echo $x; ?>`, ...) but that is not urgent.

Ping @ImAReplicant 
Let's merge the latest changes if there is any missing from https://framagit.org/ImAReplicant/freshrss-youtube

**Comment by pax0707** [comment]
Updated it yesterday.
Didn't notice any issues.

**Comment by ImAReplicant** [comment]
Hello, 
I've been browsing the various topics about the youtube extension and there seems to be a misunderstanding about the fork I made.

I never wanted to steal @kevinpapst's work, in fact I had opened an issue to ask if I could integrate the fork (https://github.com/FreshRSS/Extensions/issues/133#issue-1204483674). I thought at the time that the extension was no longer maintained.
I may have done it clumsily and I really apologize for that.

The changes I've made concern peertube videos, because without my changes peertube videos wouldn't be integrated.

In extension.php file,
line 136 :
`if (preg_match('#^https?://www\.youtube\.com/watch\?v=|/videos/watch/[0-9a-f-]{36}$#', $link) !== 1) {`
to
`if (preg_match('#^https?://www\.youtube\.com/watch\?v=|/w/[0-9a-zA-Z]{22}$#', $link) !== 1) {`

line 177 : 
`$url = str_replace('/watch', '/embed', $link);`
to
`$url = str_replace('/w', '/videos/embed', $link);`

As for the rest of the changes, it was an attempt to modify the display of the peertube description, but it doesn't seem to work anymore. Indeed, the description is displayed in a rather strange way:

<details>

Capture d’écran du 2024-03-16 10-45-30

</details>

Sorry again for this misunderstanding and the problems it has caused.

**Comment by kevinpapst** [comment]
Thanks for your explanation @ImAReplicant 

From my end this is ready for merge, as @pax0707 confirmed that the current version works.

Let's do the Peertube (how many people use that?) integration either in its own extension or at least in second PR.

**Comment by Alkarex** [comment]
Could you please update https://github.com/FreshRSS/Extensions/blob/master/repositories.json accordingly?

**Comment by kevinpapst** [comment]
Done

**Comment by kevinpapst** [comment]
Thanks for accepting the extension!

**Comment by Alkarex** [comment]
https://github.com/FreshRSS/Extensions/issues/222

**Pull request merged**